core/src/main/java/com/linecorp/armeria/common/HttpRequest.java

```diff
@@ -605,4 +606,60 @@ default HttpRequest mapError(Function<? super Throwable, ? extends Throwable> fu
         requireNonNull(function, "function");
         return of(headers(), HttpMessage.super.mapError(function));
     }
+
+    /**
+     * Applies the specified {@link Consumer} to the {@link HttpData}s
+     * emitted by this {@link HttpRequest}.
+     *
+     * <p>For example:<pre>{@code
+     * HttpRequest request = HttpRequest.of(RequestHeaders.of(HttpMethod.POST, "/items"),
+     *                                      HttpData.ofUtf8("data1,data2"));
+     * HttpRequest result = request.peekData(data -> {
+     *     assert data.toStringUtf8().equals("data1,data2");
+     * });
+     * }</pre>
+     */
+    @UnstableApi
+    default HttpRequest peekData(Consumer<? super HttpData> action) {
```

Comment:
How about moving up some common methods to `HttpMessage` such as `peekData()`, `mapData()`, `mapError()` and so on.

Reply:
Let me explain with `peekData` case!
```java
// HttpMessage.java
HttpMessage peekData(Consumer<? super HttpData> action);

// HttpRequest.java
HttpRequest peekData(Consumer<? super HttpData> action) {
  ...
  return of(headers(), stream);
}

// HttpResponse.java
HttpResponse peekData(Consumer<? super HttpData> action) {
  ...
  return of(stream);
}
```
(As you can se above code)
Cause `peekData`'s `returnType` and `of(..)` is different between `HttpRequest` and `HttpResponse`,
We can't move it on `HttpMessage` like below!
```java
// HttpMessage.java

// Possible
HttpMessage peekData(Consumer<? super HttpData> action);

// Impossible
default HttpMessage peekData(Consumer<? super HttpData> action) {
  requireNonNull(action, "action");
  final StreamMessage<HttpObject> stream = peek(action, HttpData.class);
  return of(stream); // of(..) is static method that defined on each `HttpRequest` and `HttpResponse`
}
```

> moving up some common methods to HttpMessage such as peekData(), mapData(), mapError() and so on.

So If what I understand is NOT what you meant, 
then could you explain to me with simple example code? 🙇

Reply:
> Above Case 1) is what you meant?

Yes, I meant the case 1). 😄
```java
interface HttpMessage {
    HttpMessage peekData(Consumer<? super HttpData> action);
}

interface HttpRequest {
    @Override
    default HttpRequest peekData(Consumer<? super HttpData> action) {
        ...
    }
}

interface HttpResponse {
    @Override
    default HttpResponse peekData(Consumer<? super HttpData> action) {
         ...
    }
}
```

Reply:
aha~ I got it! I updated this PR with `case 1)` 😄